src/niri.rs

```diff
@@ -2531,10 +2531,11 @@ impl Niri {
         output: &Output,
         y_invert: bool,
     ) {
-        if let Some(mut screencopy) = self
+        for mut screencopy in self
             .output_state
             .get_mut(output)
-            .and_then(|s| s.pending_screencopy.take())
+            .map(|s| s.pending_screencopy.drain(..).collect::<Vec<_>>())
```

Comment:
Uh, no extra collect() pls. mem::take()

Reply:
oh yeah. originally i thought mem::replace with default seems verbose. that's why i did this instead. mem::take is much more appropriate here, though